### PR TITLE
fix: propagate float types for ivf centroids

### PR DESCRIPTION
**Pull request by chebbyChefNEQ**

### CODE REVIEW

File: python/python/lance/dataset.py [hunk 1]
```diff
@@ -1059,12 +1059,15 @@ def create_index(
                             f"Ivf centroids must be 2D array: (clusters, dim), "
                             f"got {ivf_centroids.shape}"
                         )
-                    if ivf_centroids.dtype != np.float32:
+                    dim = ivf_centroids.shape[1]
+                    if ivf_centroids.dtype == np.float32:
+                        values = pa.array(ivf_centroids.reshape(-1), type=pa.float32())
+                    elif ivf_centroids.dtype == np.float16:
```
Comment (eddyxu) [inline python/python/lance/dataset.py:1065]: what if they are float64 or bf16?

Comment (chebbyChefNEQ) [reply python/python/lance/dataset.py:1065]: let's handle those later I guess. np doesn't have bf16 I think. fp64 probably has too little use cases in ML

Comment (eddyxu) [reply python/python/lance/dataset.py:1065]: Bfloat16 is in the ml types package from Jax. Without it, it seems to become a problem very soon ?

**Comment by wjones127** [comment]
Closing in favor of #2097

**Pull request closed without merging**